apps/api-nestjs/src/shared/validation/parse-address.pipe.ts

```diff
@@ -0,0 +1,17 @@
+import { ArgumentMetadata, BadRequestException, Injectable, PipeTransform } from '@nestjs/common'
+import { EthService } from '@app/shared/eth.service'
+
+@Injectable()
+export class ParseAddressPipe implements PipeTransform<string, string> {
+
+  constructor(private readonly ethService: EthService){}
+
+  transform(value: string, metadata: ArgumentMetadata): string {
+
+    if (!this.ethService.isValidAddress(value)) {
+      throw new BadRequestException('Invalid address hash');
```

Comment:
Is there a better way we could be throwing this exception to make it nicer to handle on the frontend?